README.md

```diff
@@ -323,8 +323,40 @@ Writing: ./db/schema.sql
 
 dbmate supports options passed to a migration block in the form of `key:value` pairs. List of supported options:
 
+- `env`
 - `transaction`
 
+**env**
+
+`env` is useful if you need to inject some environment variables into the SQL. Only environment variables explicitly listed will be available for templating:
+
+```sql
+-- migrate:up env:ADMIN_PASSWORD
+create role 'adminuser' login password '{{ .ADMIN_PASSWORD }}';
+```
+
+`env` supports multiple occurrences. If a referenced variable is not available the command will return an error:
+
+```sh
+$ dbmate up
+Applying: ...
+Error: template: tmpl:35:51: executing "tmpl" at <.ADMIN_PASSWORD>: map has no entry for key "ADMIN_PASSWORD"
+```
+
+Variables injection leverages Go's [template](https://pkg.go.dev/text/template) features. For example you can provide default values in order to prevent errors:
+
+```sql
+-- migrate:up env:THE_ROLE env:THE_PASSWORD
+create role '{{ or (index . "THE_ROLE") "adminuser" }}' login password '{{ .THE_PASSWORD }}';
+```
+
+The `js` function may help to mitigate risks of SQL injection:
```

Comment:
sorry if I missed the discussion- how does this mitigate SQL injections risks? Can't I insert a string with a single quote still?

Reply:
The [go/template functions](https://pkg.go.dev/text/template#hdr-Functions) includes `js` which is described as:

```
js
	Returns the escaped JavaScript equivalent of the textual
	representation of its arguments.
```

While this may not necessarily be sufficient to prevent *all* possible SQL injection attacks (see: Unicode homoglyph injection attacks aka Unicode smuggling [1](https://hackernoon.com/%CA%BC-%C5%9B%E2%84%87%E2%84%92%E2%84%87%E2%84%82%CA%88-how-unicode-homoglyphs-will-break-your-custom-sql-injection-sanitizing-functions-1224377f7b51) [2](https://www.dionach.com/en-us/blog/fun-with-sql-injection-using-unicode-smuggling/), character set transcoding SQL injection attacks [3](https://shiflett.org/blog/2006/addslashes-versus-mysql-real-escape-string)), it prevents the most trivial kind where an unescaped single quote `'` is in a string that will be interpolated into the SQL query, by escaping it with a backslash, thus turning `'` into `\'`.

Reply:
The trivial case of unescaped single quote is covered by [TestResolveMitigatingSqlInjection](https://github.com/amacneil/dbmate/pull/439/files#diff-e38ec788aaa9a3ec91568145000f7d7666bd296e62d65dadd1b32acc63bf9652R95).

Reply:
nice test case!

Reply:
I added a comment in the main PR thread - I'm not a fan of introducing functions to the template syntax, or encouraging use of js escaping functions for sql escaping.